File: src/enviroments/Jungle.ts
import Character from '../common/Character';
import { Environment } from '../common/Environment';

export class Jungle extends Environment {
  constructor() {
    super();
    this.name = 'Jungle';
    this.advantages = 2;
    this.disadvantages = 5;
  }

  apply(characters: Character[]): void {
    characters.forEach((character) => {
      character.speed -= (this.disadvantages * character.speed) / 100;
      character.defense += (this.advantages * character.speed) / 100;
    });
  }
}

File: environment/jungle.py


from typing import List
from abstracts.character import Character
from abstracts.environment import Environment


class Jungle(Environment):
    def __init__(self):
        super().__init__("Jungle", 2, 5)

    def apply(self, characters: List[Character]) -> None:
        for character in characters:
            character.speed -= (self.disadvantages * character.speed) / 100
            character.defense += (self.advantages * character.speed) / 100
